### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by JuanDBta**
In this branch, I started building the school library app. In this initial step, I implemented the classes to represent students and teachers.
You can find the following features:

- Created class **Person** with the following:

    - Instance vars: @id, @name, and,  @age.
    - Constructor with name, age, and, parent_permission as parameter. name and parent_permission are optional and have 
      default values of "Unknown" and true.
    - Getters for @id, @name, and,  @age.
    - Setters for @name and  @age.
    - Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
    - Public method can_use_services? that returns true if person is of age or if they have permission from parents.

Created class **Student** with the following:

    - Inherits from Person.
    - Constructor extends parent's constructor by adding @classroom and a parameter for it.
    - Method play_hooky that returns "¯\(ツ)/¯".

Created class **Teacher** with the following:

    - Inherits from Person.
    - Constructor extends parent's constructor by adding @specialization and a parameter for it.
    - Override can_use_services? so it always returns true.

Each class was saved in a separate file.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require './person'
+
+# Creating class Student
+class Student < Person
+  def initialize(age, classroom, name: 'Unknown', parent_permission: true)
+    super(name, age, parent_permission)
+    @classromm = classroom
+  end
+
+  super(name, age, parent_permission)
```
Comment (souad988) [inline student.rb]: Can you explain what this line is for here? The super(name, age, parent_permission) line outside of any method   seems to be a misplaced duplication of the constructor call.

**Review by souad988 — CHANGES_REQUESTED** [review]
Hi @JuanDBta 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

242722065-db428b0c-8dfc-49c4-a156-42f227e32868


## Highlights
Good file structure
Add default values to attributes
No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (JuanDBta) [reply student.rb]: Hey @souad988, sorry is a typo error! I will remove it. Thank you for the review!

**Review by DammyShittu — APPROVED** [review]
Hi @JuanDBta,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**